pkg/controller/job/job_controller.go

```diff
@@ -1205,7 +1205,16 @@ func (jm *Controller) enactJobFinished(job *batch.Job, finishedCond *batch.JobCo
 	if isIndexedJob(job) {
 		completionMode = string(*job.Spec.CompletionMode)
 	}
-	job.Status.Conditions = append(job.Status.Conditions, *finishedCond)
+	// append the finished condition or update if there was already a finished condition with non-true status
```

Comment:
Try using `ensureJobConditionStatus` directly. It's ok if the status is timestamps are overwritten.

Reply:
done